ssstar/src/objstore/s3.rs

```diff
@@ -582,6 +584,75 @@ impl S3Bucket {
             key
         }
     }
+
+    /// Compute what the multipart chunk size should be, based on the user's configured chunk size
+    /// but also informed by the estimated size of the object
+    fn compute_multipart_chunk_size(
+        &self,
+        key: &str,
+        size_hint: Option<u64>,
+    ) -> Result<Option<usize>> {
+        let multipart_chunk_size = self
+            .inner
+            .objstore
+            .inner
+            .config
+            .multipart_chunk_size
+            .get_bytes() as usize;
+
+        match size_hint {
+            None => {
+                // Hope that the final size of the object will be small enough that the configured
+                // chunk size is larger than 1/10,000th of the size of the whole object, but assume
+                // it will be large enough that we should use multipart
+                Ok(Some(multipart_chunk_size))
+            }
+            Some(size_hint) => {
+                if size_hint > 5 * (1024 * 1024 * 1024 * 1024u64) {
```

Comment:
What is size `5 * (1024 * 1024 * 1024 * 1024u64)` mean? Why isn't it constant?

Reply:
Made it a constant.